File: Python/7py 3/rdw.py
"""
|--------------------------------------------------------------------------
| Remove duplicate words
|--------------------------------------------------------------------------
|
"""

def remove_duplicate_words(s):
   arr = s.split(' ')
   res = []
   for i in range(len(arr)):
      if (arr[i] in res) == False:
         res.append(arr[i])
   return ' '.join(res)


def remove_duplicate_words(s):
   return ' '.join({i:0 for i in s.split()})


def remove_duplicate_words(s):
   return ' '.join(dict.fromkeys(s.split()))


def remove_duplicate_words(s):
   a=[]
   [a.append(v) for v in s.split(" ") if v not in a]
   return str(" ").join(a)


def remove_duplicate_words(s):
   return ' '.join(sorted(set(s.split()), key = s.index))


print(remove_duplicate_words("alpha beta beta gamma gamma gamma delta alpha beta beta gamma gamma gamma delta"), "alpha beta gamma delta")
print(remove_duplicate_words("my cat is my cat fat"), "my cat is fat")

File: JavaScript/7js 3/rdw.js
/*
|--------------------------------------------------------------------------
| Remove duplicate words
|--------------------------------------------------------------------------
|
*/

function removeDuplicateWords(s) {
   let arr = s.split(' ');
   let res = [];
   for (let i = 0; i < arr.length; i++) {
      if (res.includes(arr[i]) == false) {
         res.push(arr[i])
      }
   }
   return res.join(' ');
}


function removeDuplicateWords(s) {
   return [...new Set(s.split(' '))].join(' ');
}


function removeDuplicateWords(s) {
   return s.split(' ').filter((n, i, s) => {
      return s.indexOf(n) === i;
   }).join(' ');
}


function removeDuplicateWords(s) {
   return [...s.split ` `.reduce((s, w) => s.add(w), new Set())].join ` `;
}


function removeDuplicateWords(Q) {
   return Q.replace(/ ?(\S+)/g, (V, B) => Q[B] ? '' : Q[B] = V, Q = {})
}


console.log(removeDuplicateWords('alpha beta beta gamma gamma gamma delta alpha beta beta gamma gamma gamma delta'), 'alpha beta gamma delta');